joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/utils/FileUtil.scala

```diff
@@ -1,8 +1,93 @@
 package io.joern.x2cpg.utils
 
+import java.io.{IOException, File as JFile}
+import java.nio.file.{FileAlreadyExistsException, Files, Path, SimpleFileVisitor}
 import better.files.File
 
+import java.nio.file.attribute.BasicFileAttributes
+
 object FileUtil {
+  // TODO: Replace better.files with this method
+  def usingTemporaryDirectory[U](prefix: String = "")(f: Path => U): Unit = {
+    val file = Files.createTempDirectory(prefix)
+
+    try {
+      f(file)
+    } finally {
+      delete(file)
+    }
+  }
+
+  def deleteOnExit(file: Path, swallowIOExceptions: Boolean = false): Unit = {
+    try {
+      if (Files.isDirectory(file)) {
+        file.toFile.listFiles().foreach(x => deleteOnExit(x.toPath, swallowIOExceptions))
+      }
+
+      file.toFile.deleteOnExit()
+    } catch {
+      case _: IOException if swallowIOExceptions => //
+    }
+  }
+
+  def delete(file: Path, swallowIoExceptions: Boolean = false): Unit = {
+    try {
+      if (Files.isDirectory(file)) {
+        file.toFile.listFiles().foreach(x => delete(x.toPath, swallowIoExceptions))
+      }
+
+      Files.delete(file)
+    } catch {
+      case _: IOException if swallowIoExceptions => //
+    }
+  }
+
+  implicit class PathExt(p: Path) {
+    def /(child: String): Path = {
+      p.resolve(child)
+    }
+
+    def copyToDirectory(destination: Path): Unit = {
+      require(Files.isDirectory(destination), s"${destination} must be a directory")
+      copyTo(destination / p.getFileName.toString)
+    }
+
+    def copyTo(destination: Path): Unit = {
+      if (Files.isDirectory(p)) { // TODO: maxDepth?
+        Files.walkFileTree(
+          p,
+          new SimpleFileVisitor[Path] {
+            def newPath(subPath: Path): Path = destination.resolve(p.relativize(subPath))
+
+            override def preVisitDirectory(dir: Path, attrs: BasicFileAttributes) = {
+              Files.createDirectories(newPath(dir))
+              super.preVisitDirectory(dir, attrs)
+            }
+
+            override def visitFile(file: Path, attrs: BasicFileAttributes) = {
+              Files.copy(file, newPath(file))
+              super.visitFile(file, attrs)
+            }
+          }
+        )
+      } else {
+        Files.copy(p, destination)
+      }
+    }
+
+    def createIfNotExists(asDirectory: Boolean = false, createParents: Boolean = false): Unit = {
```

Comment:
`asDirectory` does not really make sense to me. We also never use it.

Reply:
I think a name that also indicates that missing parent directories are created is more suitable.

Reply:
Sorry I missed the implementation for the `asDirectory` part. Added it now. This is going to come in a bit later as there are a few places in the codebase where we do `createDirectoryIfNotExists` from `better.files` which will use the `asDirectory` flag. Renamed the function to `createWithParentsIfNotExists`

Reply:
I see ok. Like the rename.